insights/v1alpha1/types_insights.go

```diff
@@ -205,9 +278,11 @@ type HealthCheck struct {
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=4
 	TotalRisk int32 `json:"totalRisk"`
-	// advisorURI provides the URL link to the Insights Advisor.
+	// advisorURI is required field that provides the URL link to the Insights Advisor.
+	// The link must be a valid HTTPS URL and the maximum length is 2048 characters.
+	// +kubebuilder:validation:XValidation:rule=`self.matches("^https://[^\\s]+")`,message=`advisorURI must be a valid HTTPS URL (e.g., https://example.com)`
```

Comment:
CEL URL library would be better to use here

Reply:
Same as above